### ISSUE TITLE
[Question] How to invoke an Azure endpoint using Terratest?

### ISSUE THREAD

**Issue by msl0** [gruntwork-io/terratest#1441]
Hello,
I am looking to possibly interact with Azure API as part of the tests performed. I'm not very experienced in Golang, but there are methods for many Azure services. I'm curious if there is a simple way to directly interact (GET/POST) with Azure API in Terratest. Is it currently possible to access the authentication header generated by Terratest?

This is high-level context
I'm trying to use Terratest to deploy Terraform modules to verify they work (module per resource group), [trigger Azure policy scans (resource group level)](https://learn.microsoft.com/en-us/azure/governance/policy/how-to/get-compliance-data#on-demand-evaluation-scan---rest), validate the results, and delete resources. I considered two options:
- trigger Powershell script as a part of test scenario to trigger policy scan and validate results
- trigger Azure endpoints directly to trigger policy scan and get results

I would appreciate any code examples or hints

**Comment by denis256** [comment]
Hi,
I suspect that can be used the `http_helper` package to query endpoints directly

https://github.com/gruntwork-io/terratest/blob/master/test/terraform_http_example_test.go#L68

**Comment by msl0** [comment]
Hi, thanks for the reply, that's not what I was looking for. I wanted an easy way to call Azure API endpoints, without having to take care of authentication myself. I saw in the code that there are different clients for certain operations within Azure, but I was looking for something more generic...
I ended up getting around the problem by calling a Powershell script from Terratest using the `shell.RunCommandAndGetOutput()` method. It's not the prettiest solution, but it works without any problems
